src/web100srv_unit_tests.c

```diff
@@ -8,21 +9,41 @@
 
 #include "unit_testing.h"
 
-/** Starts the web100srv process. */
-pid_t start_server(int port) {
+/** Starts the web100srv process. Has a variable number of arguments, the last
+ * of which should be null. */
+pid_t start_server(int port, ...) {
+  va_list arglist;
+  int rv;
+  char *args_for_exec[256] = {"./web100srv", "--port", NULL};
+  int i = 2;  // the first index of args_for_exec that is not NULL
+  char *arg;
   pid_t server_pid;
-  int server_return_value;
+  siginfo_t server_status;
   char port_string[] = "32767";  // max port
   fprintf(stderr, "Starting the server\n");
   sprintf(port_string, "%d", port);
   if ((server_pid = fork()) == 0) {
-    execl("./web100srv", "./web100srv", "--port", port_string, NULL);
+    args_for_exec[i++] = port_string;
+    va_start(arglist, port);
+    arg = va_arg(arglist, char*);
+    while (arg != NULL) {
+      CHECK((args_for_exec[i++] = strdup(arg)) != NULL);
+      arg = va_arg(arglist, char*);
+    }
+    va_end(arglist);
+    args_for_exec[i++] = NULL;
+    execv("./web100srv", args_for_exec);
     perror("SERVER START ERROR: SHOULD NEVER HAPPEN");
     FAIL("The server should never return - it should only be killed.");
     exit(1);
   }
   // Wait until the server port (hopefully) becomes available
   sleep(1);
+  // Double-check that the server didn't crash on startup.
+  server_status.si_pid = 0;
+  rv = waitid(P_PID, server_pid, &server_status, WEXITED | WSTOPPED | WNOHANG);
+  ASSERT(rv == 0 && server_status.si_pid == 0,
```

Comment:
I don't understand the `server_status.si_pid == 0` part of this. I don't see mention in the docs of it setting this value if the process has not exited yet.